vllm/worker/model_runner.py

```diff
@@ -1046,17 +1050,21 @@ def profile_run(self) -> None:
         # Profile memory usage with max_num_sequences sequences and the total
         # number of tokens equal to max_num_batched_tokens.
         seqs: List[SequenceGroupMetadata] = []
-        # Additional GPU memory may be needed for vision encoding, which needs
-        # to be accounted for when calculating the GPU blocks for
+        # Additional GPU memory may be needed for multi-modal encoding, which
+        # needs to be accounted for when calculating the GPU blocks for
         # vLLM blocker manager.
         # To exercise the worst scenario for GPU memory consumption,
         # the number of seqs (batch_size) is chosen to maximize the number
         # of images processed.
         model_config = self.model_config
+        mm_config = self.multimodal_config
 
-        if supports_multimodal(self.model):
-            max_mm_tokens = MULTIMODAL_REGISTRY \
-                .get_max_multimodal_tokens(model_config)
+        input_registry = self.input_registry
+        mm_registry = self.mm_registry
+        mm_registry.init_mm_limits_per_prompt(model_config, mm_config)
```

Comment:
How about moving `mm_registry.init_mm_limits_per_prompt` into the model runner's `__init__` phase? As some model runners don't have a profiling run phase

Reply:
I think that's a good point - I assume this is regarding generating embeddings from a LMM? WDYT? @DarkLight1337

Reply:
Yeah, it should be fine to move it to `__init__`. Can you also implement this in #7530?

Reply:
Perhaps we need to factor out the profiling logic into its own class.

Reply:
Yea I'm doing it in #7530 @AllenDou @DarkLight1337